### PR TITLE
feat: separate providers

### PR DESCRIPTION
**Pull request by fampinheiro**
This PR includes the separation of providers so the article and profile providers don't affect one another.

This PR doesn't include:
- article loading state
- article error state
- ui changes to pagination

### CODE REVIEW

File: packages/author-profile/author-profile-content.js [hunk 1]
```diff
@@ -1,41 +1,112 @@
...
+                  onPrev={onPrev}
+                  page={page}
+                  pageSize={pageSize}
+                />
+              </View>
+            </View>
+          </View>
+        )}
+        renderSeparator={() => <AuthorProfileItemSeparator />}
+        // onEndReached={() => onNext()}
```
Comment (toboid) [inline packages/author-profile/author-profile-content.js]: left over comment

File: packages/author-profile/author-profile-content.js [hunk 2]
```diff
@@ -1,41 +1,111 @@
...
+      onPrev,
+      page,
+      pageSize,
+      onArticlePress
+    } = this.props;
+
+    const { count, dataSource } = this.state;
+
+    return (
+      <ListView
```
Comment (craigbilner) [inline packages/author-profile/author-profile-content.js]: think we need to declare `enableEmptySections`?

Comment (craigbilner) [inline packages/author-profile/author-profile-content.js]: now we've upgraded React should probably be a `FlatList`?

File: packages/author-profile/author-profile-content.web.js [hunk 3]
```diff
@@ -1,49 +1,90 @@
...
+              {separatorComponent}
+              <AuthorProfileItem
+                {...article}
+                onPress={e => onArticlePress(e, { id, url })}
+              />
+            </View>
+          );
+        })}
+    </View>
+  </View>
```
Comment (craigbilner) [inline packages/author-profile/author-profile-content.web.js]: think we need to keep the footer pagination on web?

File: packages/author-profile/author-profile.js [hunk 4]
```diff
@@ -26,67 +27,62 @@ const AuthorProfile = ({
...
-  pageSize: 10
+  pageSize: 10,
+  onTwitterLinkPress: () => {},
+  onArticlePress: () => {}
+};
+
+AuthorProfile.propTypes = {
+  isLoading: PropTypes.bool,
+  error: PropTypes.shape(),
+  author: PropTypes.shape(),
```
Comment (craigbilner) [inline packages/author-profile/author-profile.js]: can these empty shapes be more specific?

File: packages/author-profile/author-profile.js [hunk 5]
```diff
@@ -26,67 +27,62 @@ const AuthorProfile = ({
...
     return (
-      <AuthorProfileContent
-        {...data}
-        onTwitterLinkPress={onTwitterLinkPress}
-        onArticlePress={onArticlePress}
-        onNext={onNext}
-        onPrev={onPrev}
-      />
+      <ArticleListProviderWithPageState
+        articleImageRatio={"3:2"}
```
Comment (craigbilner) [inline packages/author-profile/author-profile.js]: bit sad this isn't a lint error?

Comment (fampinheiro) [reply packages/author-profile/author-profile.js]: What is the format you were expecting? `articleImageRatio="3:2"`?

File: packages/ad/package.json [hunk 6]
```diff
@@ -13,12 +13,8 @@
   "jest": {
     "preset": "react-native",
     "rootDir": "../../",
-    "transformIgnorePatterns": [
-      "node_modules/(?!@times-components)/"
-    ],
-    "testMatch": [
-      "<rootDir>/packages/ad/__tests__/**.test.js"
-    ],
+    "transformIgnorePatterns": ["node_modules/(?!@times-components)/"],
```
[the remark this answers is not in the document]
Comment (fampinheiro) [reply packages/ad/package.json]: Travis was complaining my code didn't met the formatting expectations.

File: packages/author-profile/__tests__/author-profile-helper.js [hunk 7]
```diff
@@ -6,31 +6,134 @@ import React from "react";
...
+    request: {
+      query: addTypenameToDocument(articleListQuery),
+      variables: {
+        slug: "fiona-hamilton",
+        first: 3,
+        skip: 0,
+        imageRatio: "3:2"
+      }
+    },
+    result: articlesList(0, 3)
```
[the remark this answers is not in the document]
Comment (fampinheiro) [reply packages/author-profile/__tests__/author-profile-helper.js]: 👀

File: packages/author-profile/author-profile.js (hunk 5 above)
Comment (craigbilner) [reply packages/author-profile/author-profile.js]: yah, just a string isn't it?

File: packages/ad/package.json (hunk 6 above)
[the remark this answers is not in the document]
Comment (fampinheiro) [reply packages/ad/package.json]: https://travis-ci.org/newsuk/times-components/builds/288549121?utm_source=email&utm_medium=notification

File: packages/author-profile/__tests__/author-profile-helper.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (fampinheiro) [reply packages/author-profile/__tests__/author-profile-helper.js]: This was related to the client mock. Changed the empty story to have a local mock and the issue is not happening anymore.

**Comment by fampinheiro** [comment]
The uncovered lines refer to integration with other components or testing article functionality.

https://coveralls.io/builds/13739286/source?filename=packages%2Fauthor-profile%2Fauthor-profile-content.js#L48
https://coveralls.io/builds/13739286/source?filename=packages%2Fauthor-profile%2Fauthor-profile-content.web.js#L51
https://coveralls.io/builds/13739286/source?filename=packages%2Fauthor-profile%2Fauthor-profile-content.web.js#L85

**Comment by trevorah** [comment]
> Coverage decreased (-3.0%) to 95.0%

@fampinheiro almost there!

File: packages/author-profile/__tests__/__snapshots__/author-profile-content.native.test.js.snap [hunk 8]
```diff
@@ -1,4266 +1,657 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
-exports[`renders profile 1`] = `
-<AuthorProfile
-  __typename="Author"
-  articles={
+exports[`renders profile 1`] = `<AuthorProfileLoading />`;
```
Comment (toboid) [inline packages/author-profile/__tests__/__snapshots__/author-profile-content.native.test.js.snap]: Looks like the rendered output for `renders profile 1` has changed from AuthorProfile to AuthorProfileLoading

File: packages/author-profile/__tests__/author-profile-helper.js [hunk 9]
```diff
@@ -39,13 +144,35 @@ export default AuthorProfileContent => {
...
   it("renders profile content", () => {
-    const component = renderer.create(<AuthorProfile {...props} />);
+    const component = renderer.create(
+      withMockProvider(<AuthorProfile {...props} />)
+    );
+
+    expect(component).toMatchSnapshot();
+  });
+
+  it("renders profile content", () => {
```
Comment (toboid) [inline packages/author-profile/__tests__/author-profile-helper.js]: This test has the same name as the previous one, looking at the snapshots they are testing different states

File: packages/author-profile/__tests__/author-profile-content.web.test.js [hunk 10]
```diff
@@ -1,4 +1,56 @@
...
+  );
+
+  expect(
+    component
+      .find("Pagination")
+      .at(0)
+      .props()
+      .generatePageLink(1)
+  ).toEqual("?page=1");
+  component
```
Comment (craigbilner) [inline packages/author-profile/__tests__/author-profile-content.web.test.js]: can this be a separate test, a little confusing to have this indirect assertion here?

File: packages/author-profile/__tests__/__snapshots__/author-profile-content.web.test.js.snap [hunk 11]
```diff
@@ -1,895 +1,338 @@
...
+    animating={true}
+    color="#999999"
+    hidesWhenStopped={true}
+    size="large"
+  />
+</View>
+`;
+
+exports[`renders profile content 2`] = `
+<RCTScrollView
```
Comment (craigbilner) [inline packages/author-profile/__tests__/__snapshots__/author-profile-content.web.test.js.snap]: Are we sure that the web is actually being tested here?

File: packages/author-profile/__tests__/author-profile-helper.js (hunk 9 above)
Comment (craigbilner) [reply packages/author-profile/__tests__/author-profile-helper.js]: In general I think the test names could do with a bit more detail?

**Comment by craigbilner** [comment]
@fampinheiro looks like the storybook paging is still a bit awry. Also did you put the footer pagination back?

File: packages/author-profile/__tests__/__snapshots__/author-profile-content.native.test.js.snap (hunk 8 above)
Comment (fampinheiro) [reply packages/author-profile/__tests__/__snapshots__/author-profile-content.native.test.js.snap]: 👍

File: packages/author-profile/__tests__/__snapshots__/author-profile-content.web.test.js.snap (hunk 11 above)
Comment (fampinheiro) [reply packages/author-profile/__tests__/__snapshots__/author-profile-content.web.test.js.snap]: We are sure that the ScrollView is being rendered and we know the ScrollView should be rendered only on web.

The only thing we are not testing is if the `packager` does it job and loads the `.web.js` file when we are rendering web.

**Pull request merged**